### PR TITLE
fix: `poetry env` broken on Windows (#4615)

### PR DESCRIPTION
**Pull request by tom-bowles**
Resolves: #4615

Looks like this if statement was copied from further up in the function and only one of the branches updated to use self._path.

### CODE REVIEW

**Comment by abn** [comment]
@tom-bowles can you rebase this so the cirrus tests can have a rerun please?

**Comment by tom-bowles** [comment]
@abn - Thanks for taking a look! I've rebased now.

**Comment by abn** [comment]
@tom-bowles should have mentioned this earlier, any cance we can add in a test case for this so we do not regress?

**Comment by tom-bowles** [comment]
@abn - no worries, I'll have a crack at that today

**Comment by tom-bowles** [comment]
@abn - Added test.

**Review by neersighted — CHANGES_REQUESTED** [review]
LGTM -- we just need the formatting fixed (run `pre-commit run --all-files`) by black to merge.

File: tests/utils/test_env.py [hunk 1]
```diff
@@ -1001,6 +1001,27 @@ def test_env_finds_the_correct_executables(tmp_dir, manager):
...
+
+    if bin_dir_path.exists():
+        if base_dir_path.exists():
+            bin_dir_path.unlink()
+        else:
+            bin_dir_path.rename(base_dir_path)
+
+    venv = VirtualEnv(venv_path)
+
+    assert Path(venv.python) == base_dir_path
```
Comment (neersighted) [inline tests/utils/test_env.py:1023]: Looks like this test fails on Python 3.6 on Windows, so I spoke a little too soon. Can we take a look at that failure and get it corrected?

**Comment by dimbleby** [comment]
Looks like this was fixed incidentally at #5007.

Apparently that one was allowed through without testcases so I suppose it could _just about_ be worth keeping this open if anyone is motivated, more than a year later, to add a unit test.

But I expect this should now be closed.

**Pull request closed without merging**